### PR TITLE
chore: Bump golangci-lint to v1.53.2

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/kind support

**What this PR does / why we need it**:
bump golangci-lint to v1.53.2 and fix errors related to that.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Comments _Originally posted by @richardcase in https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4321#pullrequestreview-1471233824_ addressed. For one of the comments: _Originally posted by @Ankitasw in https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4321#issuecomment-1584001902_

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Bump golangci-lint to v1.53.2
```

### CODE REVIEW

**Comment by Ankitasw** [comment]
PR is ready fro review.

File: exp/controllers/awsmachinepool_controller.go [hunk 1]
```diff
@@ -214,14 +215,14 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(ctx context.Context, machineP
 	if !machinePoolScope.Cluster.Status.InfrastructureReady {
 		machinePoolScope.Info("Cluster infrastructure is not ready yet")
 		conditions.MarkFalse(machinePoolScope.AWSMachinePool, expinfrav1.ASGReadyCondition, infrav1.WaitingForClusterInfrastructureReason, clusterv1.ConditionSeverityInfo, "")
-		return ctrl.Result{}, nil
+		return ctrl.Result{Requeue: true}, nil
```
Comment (enxebre) [inline exp/controllers/awsmachinepool_controller.go:218]: why is this returning true now? that would active the rate limiter, wouldn't this be just requeue when a change in the status triggers and event?

File: exp/controllers/awsmachinepool_controller.go [hunk 2]
```diff
@@ -214,14 +215,14 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(ctx context.Context, machineP
...
 	}
 
 	// Make sure bootstrap data is available and populated
 	if machinePoolScope.MachinePool.Spec.Template.Spec.Bootstrap.DataSecretName == nil {
 		machinePoolScope.Info("Bootstrap data secret reference is not yet available")
 		conditions.MarkFalse(machinePoolScope.AWSMachinePool, expinfrav1.ASGReadyCondition, infrav1.WaitingForBootstrapDataReason, clusterv1.ConditionSeverityInfo, "")
-		return ctrl.Result{}, nil
+		return ctrl.Result{RequeueAfter: time.Minute}, nil
```
Comment (enxebre) [inline exp/controllers/awsmachinepool_controller.go:225]: same question than https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4326/files#r1244893865

**Comment by enxebre** [comment]
dropped a few questions lgtm otherwise

File: exp/controllers/awsmachinepool_controller.go (hunk 1 above)
Comment (Ankitasw) [reply exp/controllers/awsmachinepool_controller.go:218]: Done it based on discussion here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4321#issuecomment-1588846160

**Comment by Ankitasw** [comment]
@richardcase can you take a look at this?

**Pull request closed without merging**